arelle/oim/Load.py

```diff
@@ -1759,6 +1760,22 @@ def dimChecks():
                 initialComment="extracted from OIM {}".format(mappedUri),
                 base=documentBase)
             _return = modelXbrl.modelDocument
+
+        # Updating the namespace map on the document is expensive,
+        # and increases cost as the size of the document grows.
+        # To minimize the effect of this, we can add any known and valid
+        # prefix/namespace combinations to the document preemptively.
+        for prefix, namespace in namespaces.items():
+            if not prefix or not namespace:
+                continue
+            # First, test if the prefix/namespace combination is valid
```

Comment:
Out of curiosity, what combinations would be invalid?  Are invalid prefixes checked elsewhere, and if so, can we make this code only run on the validated ones?

Reply:
I think we should be able to do that [here](https://github.com/Arelle/Arelle/blob/173c273051ab65dbda0794584d2e8ae01fb90f6f/arelle/oim/Load.py#L747-L766). Though I suspect the regex validations aren't as robust as they'd need to be.

Reply:
That seems like a gap in the OIM test suite (should be using such a URI) and a gap in Arelle (should be encoding the strings so that they are valid URIs, per [OIM](https://www.xbrl.org/Specification/oim-common/PR-2021-08-04/oim-common-PR-2021-08-04.html#sec-uri-map) -> https://www.w3.org/TR/xmlschema-2/#anyURI -> [xlink](https://www.w3.org/TR/2001/REC-xlink-20010627/#link-locators)) that should be fixed rather than just ignoring the prefixes?

Reply:
The testcases in question are around the `oimce:multipleAliasesForURI` error, which it seems is less concerned about the validity of URIs being used and more about whether or not they are duplicate URIs. It introduces the invalid URI above and confirms that we don't consider it a duplicate of "http://xbrl.org/entity/identification/sch%20eme" since uniqueness is determined using the raw values.

This is the testcase description: 
> Two different aliases are bound to URIs that are the same after applying the XLink URI encoding. We use string comparison of raw values when determining uniqueness.

Even then, the prefixes aren't necessarily *ignored*, they're just not included in this optimization. If the prefix/namespace in question is actually used and Arelle attempts to add it to the document later on, then it should likely trigger `oimce:invalidURI`.